src/jit/gentree.h

```diff
@@ -2673,20 +2694,62 @@ struct GenTreeArgList : public GenTreeOp
     {
     }
 
-    GenTreeArgList(GenTreePtr arg, GenTreeArgList* rest) : GenTreeOp(GT_LIST, TYP_VOID, arg, rest)
+    GenTreeArgList(GenTreePtr arg, GenTreeArgList* rest) : GenTreeArgList(GT_LIST, arg, rest)
     {
-        // With structs passed in multiple args we could have an arg
-        // GT_LIST containing a list of LCL_FLDs, see IsListForMultiRegArg()
-        //
-        assert((arg != nullptr) && ((!arg->IsList()) || (arg->IsListForMultiRegArg())));
+    }
+
+    GenTreeArgList(genTreeOps oper, GenTreePtr arg, GenTreeArgList* rest) : GenTreeOp(oper, TYP_VOID, arg, rest)
+    {
+        assert(OperIsAnyList(oper));
+        assert((arg != nullptr) && arg->IsValidCallArgument());
         gtFlags |= arg->gtFlags & GTF_ALL_EFFECT;
         if (rest != nullptr)
         {
             gtFlags |= rest->gtFlags & GTF_ALL_EFFECT;
         }
     }
+};
+
+struct GenTreeFieldList : public GenTreeArgList
+{
+    unsigned  gtFieldOffset;
+    var_types gtFieldType;
+    bool      gtIsListHead;
```

Comment:
Yes - good catch.

---
In reply to: [79511435](https://github.com/dotnet/coreclr/pull/7252#discussion_r79511435) [](ancestors = 79511435)